enhancements/microshift/disabling-storage-components.md

```diff
@@ -0,0 +1,239 @@
+- Do not make backwards-incompatible changes to the MicroShift config API.
+- Do not endanger or make inaccessible persistent data on upgraded clusters.
+
+### Non-Goals
+
+- Provide a generalized framework to support potential future alternatives to LVMS.
+- Generically enable installing CSI components without enabling LVMS.
+
+## Proposal
+
+### Workflow Description
+
+**_Installation with CSI Driver and Snapshotting (Default)_**
+
+1. User installs MicroShift. A MicroShift and LVMD config are not provided.
+2. MicroShift starts and detects no config and falls back to default MicroShift configuration (LVMS and CSI on by default).
+3. MicroShift checks to determine if host is compatible with LVMS. If it is not, an error is logged, 
+LVMS and CSI installation is skipped and install continues. (This it the current behavior in MicroShift)
+4. MicroShift attempts to dynamically create generate the LVMD config. If it cannot, an error is logged, LVMS and CSI
+installation is skipped and install continues.
+5. If all checks pass, LVMS and CSI are deployed.
+
+**_Installation without CSI Driver and Snapshotting_**
+
+1. User installs MicroShift.
+2. User specifies a MicroShift config sets fields to disable LVMS and CSI Snapshots.
+3. MicroShift starts and detects the provided config.
+4. LVMS and CSI snapshot components are not deployed.
+
+**_Post-Start Installation, LVMS only_**
+
+1. User has already installed and started MicroShift service.
+2. User later determines there is a requirement for persistent storage, but not snapshotting.
+3. User edits the MicroShift config, enabling LVMS only.
+4. User restarts MicroShift service.
+5. MicroShift starts and detects the provided config.
+6. MicroShift performs LVMS startup checks.
+7. If all checks pass, LVMS is deployed. Else if checks fail, an error is logged, LVMS deployment is skipped, 
+and startup continues.
+
+**_Complete Uninstallation, with Data removal_**
+
+> NOTE: Installation is not reversible. Deleting LVMS or the CSI snapshotter while there are still storage volumes risks
+orphaning volumes. It is the user's responsibility to manually destroy data volumes and/or snapshots before
+uninstalling.
+
+1. User has already deployed a cluster with LVMS and CSI Snapshotting installed.
+2. User has deployed cluster workloads with persistent storage.
+3. User decides that LVMS and snapshotting are no longer needed.
+4. User takes actions to back up, wipe, or otherwise ensure data will not be irretrievable.
+5. User stops workloads with mounted storage volumes.
+6. User deletes VolumeSnapshots and waits for deletion of VolumeSnapshotContent objects to verify. The deletion process
+cannot happen after the CSI Snapshotter is deleted.
+7. User delete PersistentVolumeClaims and waits for deletion of PersistentVolumes. The deletion process
+cannot happen after LVMS is deleted.
+8. User deletes the relevant LVMS and Snapshotter cluster API resources:
```

Comment:
In our `startIngressController` we have a logic to remove the Router if user sets it to "Removed" (other option in "Managed"). Can we do the same? Remove topolvm if `.storage.driver` is `none`?
Or do you think risk is too big to allow this kind flow?

Reply:
The router had this same pattern because a user had to be able to turn it off. Would the same apply here?
A user may delete the router manually, but on the next restart MicroShift had to know not to create it again. You could have that kind of state, but it was easier to just move it entirely into the configuration and have the user manage through config+restarts whether the router is there or not.

Reply:
The issue here is that if there are LVM volumes floating around, they'll be orphaned. K8s doesn't know how to cleanup the volumes without the lvms driver.